salt/states/file.py

```diff
@@ -456,6 +481,11 @@ def _check_directory(name,
     Check what changes need to be made on a directory
     '''
     changes = {}
+    if recurse or clean:
+        walk_l = list(os.walk(name))  # walk path only once and store the result
+        # root: (dirs, files) structure, compatible for python2.6
+        walk_d = dict((i[0], (i[1], i[2])) for i in walk_l)
```

Comment:
We try to avoid these types of dense comprehensions. Would you mind just splitting these out?